### ISSUE TITLE
PLNRulesUTest does not pass.

### ISSUE THREAD

**Issue by linas** [opencog/opencog#2487]
For me, PLNRulesUTest has not been passing for many months or longer.  Apparently, this is an abnormal condition.

I get these errors:
```
Running cxxtest tests (19 tests)[2016-11-19 09:53:42:031] [DEBUG] BEGIN TEST: test_deduction
[2016-11-19 09:53:42:131] [ERROR] Backtrace:
           8 (apply-smob/1 #<catch-closure 55946f633c00>)
           7 (apply-smob/1 #<catch-closure 55946f633b80>)
In ice-9/boot-9.scm:
   2327:4  6 (save-module-excursion #<procedure 55946f634d40 at ice-…>)
In ice-9/eval-string.scm:
     38:6  5 (read-and-eval #<input: string 55946f9bf770> #:lang _)
In ice-9/eval.scm:
   191:35  4 (_ #f)
   223:20  3 (proc #<directory (guile-user) 55946f695750>)
In unknown file:
           2 (%resolve-variable (7 . deduction-inheritance-rule) #<d…>)
In ice-9/boot-9.scm:
   750:24  1 (dispatch-exception _ unbound-variable (#f "Unbound …" …))
In unknown file:
           0 (apply-smob/1 #<catch-closure 55946f633b40> # #f "Unbo…" …)

ERROR: In procedure apply-smob/1:
ERROR: Unbound variable: deduction-inheritance-rule
```

and this:
```
59: Error: Test failed: Backtrace:
           8 (apply-smob/1 #<catch-closure 55946f633c00>)
           7 (apply-smob/1 #<catch-closure 55946f633b80>)
In ice-9/boot-9.scm:
   2327:4  6 (save-module-excursion #<procedure 55946f634d40 at ice-…>)
In ice-9/eval-string.scm:
     38:6  5 (read-and-eval #<input: string 55946f9bf770> #:lang _)
In ice-9/eval.scm:
   191:35  4 (_ #f)
   223:20  3 (proc #<directory (guile-user) 55946f695750>)
In unknown file:
           2 (%resolve-variable (7 . deduction-inheritance-rule) #<d…>)
In ice-9/boot-9.scm:
   750:24  1 (dispatch-exception _ unbound-variable (#f "Unbound …" …))
In unknown file:
           0 (apply-smob/1 #<catch-closure 55946f633b40> # #f "Unbo…" …)

ERROR: In procedure apply-smob/1:
ERROR: Unbound variable: deduction-inheritance-rule
ABORT: unbound-variable
```

and this:

```
2016-11-19 09:53:42:333] [DEBUG] BEGIN TEST: test_implication_full_instantiation
[2016-11-19 09:53:42:441] [ERROR] Backtrace:
           7 (apply-smob/1 #<catch-closure 55946fba4540>)
           6 (apply-smob/1 #<catch-closure 55946fba44c0>)
In ice-9/boot-9.scm:
   2327:4  5 (save-module-excursion #<procedure 55946fc4f360 at ice-…>)
In ice-9/eval-string.scm:
     38:6  4 (read-and-eval #<input: string 55946fb27700> #:lang _)
In ice-9/eval.scm:
   223:20  3 (proc #<directory (guile-user) 55946f695750>)
In unknown file:
           2 (%resolve-variable (7 . #) #<directory (guile-user) 559…>)
In ice-9/boot-9.scm:
   750:24  1 (dispatch-exception _ unbound-variable (#f "Unbound …" …))
In unknown file:
           0 (apply-smob/1 #<catch-closure 55946fba4480> # #f "Unbo…" …)

ERROR: In procedure apply-smob/1:
ERROR: Unbound variable: implication-full-instantiation-rule
```

and a bunch more. One for this:
```
 Unbound variable: implication-partial-instantiation-rule
```
concludes:
```
Failed 4 and Skipped 0 of 19 tests
Success rate: 78%
```

**Comment by linas** [comment]
after spending many hours on this, the core bug seems to have been due to a guile caching problem; some of the PLN rules files had changed, but guile did not realize this, and used the old versions of the files.  

The confusion was made considerably worse by the fact that there are two copies of `implication-instantiation-rule.scm` in the repos: one in the atomspace repo, in `tests/rule-engine/rules` and one in the opencog repo. `opencog/pln/rules`  During debugging, I think I was able to make it pick up the other version of this file, which led to unexplainable directory-dependence bugs.

I'm closing for now, as I'm not sure what to do, or where the real bug was.  At the moment, I can no longer reproduce it.  Waste of time.
